### PR TITLE
Path for run_sorters with docker images

### PR DESCRIPTION
**Pull request by yger**
We would like to be able to use run_sorters with docker images for some of the sorters. This should deal with that. The run_sorters command also accept docker_images = {sorter_name : docker_image}. If provided, adequate docker image are used instead of local sorters

### CODE REVIEW

File: spikeinterface/sorters/launcher.py [hunk 1]
```diff
@@ -161,13 +167,20 @@ def run_sorters(sorter_list,
                         shutil.rmtree(str(output_folder))
 
             params = sorter_params.get(sorter_name, {})
+            use_docker = sorter_name in docker_images
+            if use_docker:
+                docker_image = docker_images[sorter_name]
+            else:
+                docker_image = None
+
```
Comment (samuelgarcia) [inline spikeinterface/sorters/launcher.py:175]: Can be done with in one line:

`docker_image = docker_images.get(sorter_name, None)`

File: spikeinterface/sorters/launcher.py [hunk 2]
```diff
@@ -50,6 +55,7 @@ def run_sorters(sorter_list,
                 engine_kwargs={},
                 verbose=False,
                 with_output=True,
+                docker_images = {}
```
Comment (samuelgarcia) [inline spikeinterface/sorters/launcher.py:58]: also extend the dostring

**Comment by samuelgarcia** [comment]
Cool.

Could you also make a small test that is skiped by default for GH actions but that could be run and tested on local machine ?

File: spikeinterface/sorters/launcher.py (hunk 1 above)
Comment (yger) [reply spikeinterface/sorters/launcher.py:175]: Done

File: spikeinterface/sorters/launcher.py (hunk 2 above)
Comment (yger) [reply spikeinterface/sorters/launcher.py:58]: Done

**Comment by yger** [comment]
I'll have a look, not sure of how to do that however...

**Comment by samuelgarcia** [comment]
Have a look to spikeinterface/sorters/test/test_launcher.py and a function that have the decorator
`@pytest.mark.skipif(True, reason='No run on CI')`
but the function locally test on your local computer the concept with a given image.

**Review by alejoe91 — APPROVED** [review]

**Comment by alejoe91** [comment]
@samuelgarcia @yger looks good to me! We can add the local test later

**Comment by yger** [comment]
Then merge whenever you want. I'm also discovering a bug now in recent spikeinterface with PhyKiloSortExtractor.... There is an issue if no .tsv file are there, which is the case for me.  Did you changed something there in new SI ?

**Comment by yger** [comment]
More precisely cluster_info["id"] = cluster_info["cluster_id"] triggers and error since cluster_info is None in my case (no .tsv in the result folder of KS1). I'm about to rerun all our sims, with updated SI and YASS :( I'll make a dirty patch

**Comment by alejoe91** [comment]
@yger do you have csv instead?

**Comment by yger** [comment]
Nope, no .csv or .tsv. This was I think the default, as long as you do not manually curate the data. And this is why the extractor is crashing

**Comment by yger** [comment]
Patch for KSExtractors is included in this PR, as we discussed it together. Right now everything seems to work with YASS, sounds good to me

File: spikeinterface/extractors/phykilosortextractors.py [hunk 3]
```diff
@@ -48,7 +48,12 @@ def __init__(self, folder_path, exclude_cluster_groups=None, keep_good_only=Fals
         # try to load cluster info
         cluster_info_files = [p for p in phy_folder.iterdir() if p.suffix in ['.csv', '.tsv']
                               and "cluster_info" in p.name]
-        if len(cluster_info_files) == 1:
+
+        if len(cluster_info_files) == 0:
```
Comment (samuelgarcia) [inline spikeinterface/extractors/phykilosortextractors.py:52]: This is strange. Why in a particular case this file is not present ?
Is it version related ?

If not exists I am not sure the `unsorted` is the good naming by default.

Comment (alejoe91) [reply spikeinterface/extractors/phykilosortextractors.py:52]: @samuelgarcia yes this is for ks1 and unsorted is the default name

**Pull request merged**